communications/mqtt/ESP32_MQTT/ESP32_MQTT.ino

```diff
@@ -0,0 +1,96 @@
+#include <WiFi.h>
+#include <PubSubClient.h>
+
+
+// Update these with values suitable for your network.
+const char* ssid = "WIFI_SSID";
+const char* password = "WIFI_Passw";
+const char* mqtt_server = "Server_IP";
+#define mqtt_port 1883
+#define MQTT_USER "User"
+#define MQTT_PASSWORD "Password"
+#define MQTT_SERIAL_PUBLISH_CH "MQTT/PUB/topic"
+#define MQTT_SERIAL_RECEIVER_CH "MQTT/SUB/topic"
+
+WiFiClient wifiClient;
+
+PubSubClient client(wifiClient);
+
+void setup_wifi() {
+    delay(10);
+    // We start by connecting to a WiFi network
+    Serial.println();
+    Serial.print("Connecting to ");
+    Serial.println(ssid);
+    WiFi.begin(ssid, password);
+    while (WiFi.status() != WL_CONNECTED) {
+      delay(500);
+      Serial.print(".");
+    }
+    randomSeed(micros());
+    Serial.println("");
+    Serial.println("WiFi connected");
+    Serial.println("IP address: ");
+    Serial.println(WiFi.localIP());
+}
+
+void reconnect() {
+  // Loop until we're reconnected
+  while (!client.connected()) {
+    Serial.print("Attempting MQTT connection...");
+    // Create a random client ID
+    String clientId = "ESP32Client-";
+    clientId += String(random(0xffff), HEX);
```

Comment:
I would use the `WiFi.macAddress()` which is unique and it is more consistent for device tracking instead of random